### DISCUSSION TITLE
Copilot not giving whole function suggestions and only line-by-line ones

### DISCUSSION THREAD

**Discussion by sidm1983** [community/community#10948] [Copilot Conversations]
@davecheney as discussed, I am starting a new discussion for my copilot issue.

Every time I type code, I can see the little copilot icon in the status bar change to a spinning loader and then when it is done, I still get no suggestions. I was following the first suggestion code from the copilot getting started readme (i.e. `function getDaysBetweenDates(begin, end) {`), but I didn't get a suggestion.

Here is my Github Copilot output:
```
[INFO] [default] [2022-02-03T07:46:57.792Z] [fetchCompletions] engine https://copilot-proxy.githubusercontent.com/v1/engines/copilot-aatest
[INFO] [default] [2022-02-03T07:46:58.021Z] request.response: [https://copilot-proxy.githubusercontent.com/v1/engines/copilot-aatest/completions] took 228 ms
[INFO] [default] [2022-02-03T07:46:58.021Z] prompt: {"prefix":"// Language: javascript\nfunction getDaysBetweenDates(begin, end) {}","suffix":""}
[INFO] [streamChoices] [2022-02-03T07:46:58.022Z] solution 0 returned. finish reason: ["stop"] finishOffset: [undefined] completionId: [{cmpl-4XR7Reqaau2gR5Z6gZ1Okmp1O1JZ3}] created: [{1643874417}]
[INFO] [ghostText] [2022-02-03T07:46:58.022Z] Breaking, no choices
```

I then tried adding a comment and then it started giving me suggestions, but line by line suggestions that I had to accept one at a time before it would give me the next one. Is it meant to work like that? Any insights would be greatly appreciated. Will continue to test and report back any other findings.

Thank you!

**Comment by sidm1983** [comment]
@davecheney so I have been doing some more testing today and it seems to be giving me full function suggestions today. Having said that, it does give me duplicate suggestions even though I already have a function it previously suggested to me in the same file (see screenshot below).
![image](https://user-images.githubusercontent.com/45752844/152480312-0f3cd3ba-6d31-4923-a730-495bd9c926ac.png)

**Reply by davecheney** [reply]
That's somewhat expected, copilot mostly (but not exclusively) looks at the code preceding the cursor.

**Reply by sidm1983** [reply]
Oh interesting! Thanks for the insight. Good to know.

**Comment by sidm1983** [comment]
@davecheney  So I just tested that theory and it looks like it still suggests duplicates even if I already have the same function directly preceeding it.
![image](https://user-images.githubusercontent.com/45752844/152482408-a9d61121-e233-4e41-986c-b9320ce99bc3.png)

**Reply by sidm1983** [reply]
@davecheney, what is also worth noting (and maybe correcting?) about the suggestion above is that the age test should have asserted that the employee is 32 and not 28. Someone born in 1990 would be 32 today and not 28. I also know that the test is fragile as it will stop working a certain amount of time, but thought it might be worth pointing out.

**Reply by davecheney** [reply]
Copilot isn’t perfect, it doesn’t actually understand the code it suggests.

**Reply by sidm1983** [reply]
Yup no worries. I get it. Was just pointing it out in case it was calculating that age of 28 based on the wrong year, but sounds like it might have been a randomly selected age. All good :)

Copilot is definitely amazing and a huge timesaver. I will make sure to double-check its suggestions. Thank you.
